tests/test_store/test_fsspec.py

```diff
@@ -244,3 +258,27 @@ def test_no_wrap_async_filesystem():
 
     assert not isinstance(store.fs, AsyncFileSystemWrapper)
     assert store.fs.async_impl
+
+
+@pytest.mark.skipif(
+    parse_version(fsspec.__version__) < parse_version("2024.12.0"),
+    reason="No AsyncFileSystemWrapper",
+)
+async def test_delete_dir_wrapped_filesystem(tmpdir) -> None:
+    """The local fs is not async so we should expect it to be wrapped automatically"""
+    from fsspec.implementations.asyn_wrapper import AsyncFileSystemWrapper
+
+    store = FsspecStore.from_url(tmpdir / "test/path", storage_options={"auto_mkdir": True})
```

Comment:
> tmpdir / "test/path"

This is not a URL. I don't know what from_url demands, but `f"{tmpdir}/test/path"` seems better to me.

Reply:
I used that method since it wraps the filesystem automatically, and figured that would be the entry point most people try first. FWIW, it seems to work with a local path.  I could construct the filesystem manually if preferrable, e.g., 
```python
sync_fs = LocalFileSystem(auto_mkdir=True)
async_fs = AsyncFileSystemWrapper(sync_fs)
store = FsspecStore(async_fs, read_only=False, path=f"{tmpdir}/test/path")
```

Reply:
changed in https://github.com/zarr-developers/zarr-python/pull/2661/commits/f1d90cc53f14aa317729b499364bb9d5dc70b6ad